Packet++/header/FtpLayer.h

```diff
@@ -0,0 +1,458 @@
+			RMD  = ('R') | ('M' << 8) | ('D' << 16),
+			/// Remove a directory tree
+			RMDA = ('R') | ('M' << 8) | ('D' << 16) | ('A' << 24),
+			/// Rename from.
+			RNFR = ('R') | ('N' << 8) | ('F' << 16) | ('R' << 24),
+			/// Rename to.
+			RNTO = ('R') | ('N' << 8) | ('T' << 16) | ('O' << 24),
+			/// Sends site specific commands to remote server (like SITE IDLE 60 or SITE UMASK 002). Inspect SITE HELP output for complete list of supported commands.
+			SITE = ('S') | ('I' << 8) | ('T' << 16) | ('E' << 24),
+			/// Return the size of a file.
+			SIZE = ('S') | ('I' << 8) | ('Z' << 16) | ('E' << 24),
+			/// Mount file structure.
+			SMNT = ('S') | ('M' << 8) | ('N' << 16) | ('T' << 24),
+			/// Use single port passive mode (only one TCP port number for both control connections and passive-mode data connections)
+			SPSV = ('S') | ('P' << 8) | ('S' << 16) | ('V' << 24),
+			/// Returns information on the server status, including the status of the current connection
+			STAT = ('S') | ('T' << 8) | ('A' << 16) | ('T' << 24),
+			/// Accept the data and to store the data as a file at the server site
+			STOR = ('S') | ('T' << 8) | ('O' << 16) | ('R' << 24),
+			/// Store file uniquely.
+			STOU = ('S') | ('T' << 8) | ('O' << 16) | ('U' << 24),
+			/// Set file transfer structure.
+			STRU = ('S') | ('T' << 8) | ('R' << 16) | ('U' << 24),
+			/// Return system type.
+			SYST = ('S') | ('Y' << 8) | ('S' << 16) | ('T' << 24),
+			/// Get a thumbnail of a remote image file
+			THMB = ('T') | ('H' << 8) | ('M' << 16) | ('B' << 24),
+			/// Sets the transfer mode (ASCII/Binary).
+			TYPE = ('T') | ('Y' << 8) | ('P' << 16) | ('E' << 24),
+			/// Authentication username.
+			USER = ('U') | ('S' << 8) | ('E' << 16) | ('R' << 24),
+			/// Change to the parent of the current working directory
+			XCUP = ('X') | ('C' << 8) | ('U' << 16) | ('P' << 24),
+			/// Make a directory
+			XMKD = ('X') | ('M' << 8) | ('K' << 16) | ('D' << 24),
+			/// Print the current working directory
+			XPWD = ('X') | ('P' << 8) | ('W' << 16) | ('D' << 24),
+			///
+			XRCP = ('X') | ('R' << 8) | ('C' << 16) | ('P' << 24),
+			/// Remove the directory
+			XRMD = ('X') | ('R' << 8) | ('M' << 16) | ('D' << 24),
+			///
+			XRSQ = ('X') | ('R' << 8) | ('S' << 16) | ('Q' << 24),
+			/// Send, mail if cannot
+			XSEM = ('X') | ('S' << 8) | ('E' << 16) | ('M' << 24),
+			/// Send to terminal
+			XSEN = ('X') | ('S' << 8) | ('E' << 16) | ('N' << 24)
+		};
+
+		/** A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		FtpRequestLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet) : FtpLayer(data, dataLen, prevLayer, packet) { m_Protocol = FTP; };
+
+		/**
+		 * Empty c'tor
+		 */
+		FtpRequestLayer();
```

Comment:
Maybe this c'tor shouldn't be empty and accept the command and (optional) argument as parameters?
There is no point in an FTP message that doesn't have a command so it makes sense to "force" the user to supply it in the c'tor. The argument is optional but we should also enable to pass it for easier creation of this layer